pycbc/events/ranking.py

```diff
@@ -299,6 +299,21 @@ def get_newsnr_sgveto_psdvar_scaled_threshold(trigs):
     'newsnr_sgveto_psdvar_scaled_threshold': get_newsnr_sgveto_psdvar_scaled_threshold,
 }
 
+# Lists of datasets required in the trigs object for each function
+required_datasets = {}
```

Comment:
Does it make this linked to the function themselves? I'm just wondering if this is really maintainable.

Reply:
These are not linked, the dictionary will need updating as more functions are added into this module.

One way to add this would be if we define the dictionary before the functions, and all functions have an assert that the required_datasets are in trigs. Then if another function is made (likely copied and adapted), then people will naturally add the list into this dictionary. This doesn't make it more maintainable, but does add a psychological incentive to add the list

Reply:
I've added the check, so anyone using `get_sngls_ranking_from_trigs` without adding the required datasets will have a problem

Reply:
I've had to abandon this idea, as the `get_sngls_ranking_from_trigs` function is used by quite a few different input classes for `trigs`, so making the check general enough was over-complicated (in my mind at least)